pkg/kv/kvserver/gc/gc.go

```diff
@@ -309,9 +317,12 @@ func Run(
 	if err != nil {
 		return Info{}, err
 	}
-	err = processReplicatedRangeTombstones(ctx, desc, snap, now, newThreshold, gcer, &info)
-	if err != nil {
-		return Info{}, err
+	// Fast path means all range was removed and we don't need to collect range keys.
```

Comment:
This is sort of true, but only because we don't currently write range keys across local keyspace. The fast path only applies to global keys. Nothing really prevents us from writing MVCC range tombstones across local keyspace. Should we call into this path anyway?

Reply:
We can just pass down if we want to exclude spans that we covered by clear range. Currently it is only to userspan but if we extend range keys and clear range to others we might need to split iteration over keys which was done in unfinished batch pr 😢

Reply:
Passing down user spans seems fine.